Ensure broadcasted functions retain their bounds checks

c42f opened the pull request:
The inbounds assertion within the map! implementation, combined with
propagate_inbounds on gettokenvalue(d::BroadcastedDictionary, t) leads
to removing the bound check on any user-defined function which is
broadcasted. This adds an extra layer of dispatch to protect the
user-defined function which is broadcasted.

Fixes #63

andyferris reviewed (approved):
Haha awesome - thanks for finding this one.

(Note: the original intent for `_f(d)` was to be an alias for `d.f` in the case that I overload `getproperty` on `AbstractDict` (as a shortcut for dictionaries with `Symbol` keys) so perhaps it could be renamed?)

c42f commented:
> Note: the original intent for _f(d) was to be an alias for d.f in the case that I overload getproperty on AbstractDict (as a shortcut for dictionaries with Symbol keys) so perhaps it could be renamed

Yeah I figured, but I was lazy :laughing:. I renamed it to `_call_f` and kept `_f` as it was.

Also fixed the tests by making sure `only` is defined even on older versions of Julia.

andyferris commented:
Great! 👍

There seems to be an UndefVarError in the CI.

c42f commented:
> UndefVarError in the CI

This was weird. I think it may have been something to do with `only` being defined in an inner scope in an `if` block.

Anyway, it turns out that those tests always passed regardless because CI runs with `--bounds-check=yes`.  So I move that test into its own file which can be run without the `--bounds-check` argument.

andyferris commented:
> Anyway, it turns out that those tests always passed regardless because CI runs with --bounds-check=yes. So I move that test into its own file which can be run without the --bounds-check argument.

Duh! :man_facepalming: Nice solution.

c42f commented:
> Nice solution.

I stole it from Base :laughing: 

i think it might be nicer if the whole set of tests ran in CI both with and without `--bounds-check=yes`, but it turns out this isn't so easy to do without a certain amount of hacking. https://github.com/julia-actions/julia-runtest/pull/46 should make it a lot easier.
